holmes/plugins/toolsets/grafana/loki_api.py

```
@@ -59,7 +59,15 @@ def execute_loki_query(
         return []
 
     except requests.exceptions.RequestException as e:
-        raise Exception(f"Failed to query Loki logs: {str(e)}")
+        error_details = (
+            f"Failed to query Loki logs:\n"
+            f"  URL: {url}\n"
+            f"  Query: {query}\n"
+            f"  Time range: {start} to {end}\n"
+            f"  Limit: {limit}\n"
+            f"  Error: {str(e)}"
```

Ruff (0.13.1)

Use explicit conversion flag

Replace with conversion flag

(RUF010)